src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs

```diff
@@ -0,0 +1,282 @@
+            IEnumerable<PolicyConfiguration> tfsPolicies,
+            IEnumerable<PolicyType> types)
+        {
+            List<int> handledTfsPolicies = new List<int>();
+
+            foreach (var currentPolicy in relevantPolicies)
+            {
+                this.HandleBranchChanges(policyClient, projectId, repository, tfsPolicies, types, handledTfsPolicies, currentPolicy);
+            }
+
+            this.RemoveNoMatchTfsPolicies(allowDeletion, policyClient, projectId, tfsPolicies, handledTfsPolicies);
+        }
+
+        private void HandleBranchChanges(
+            PolicyHttpClient policyClient,
+            Guid projectId,
+            GitRepository repository,
+            IEnumerable<PolicyConfiguration> tfsPolicies,
+            IEnumerable<PolicyType> types,
+            List<int> handledTfsPolicies,
+            BranchPolicies currentPolicy)
+        {
+            foreach (var policy in currentPolicy.Policies)
+            {
+                bool hasMatch = false;
+
+                var typeId = policy.Type.GetPolicyTypeId(types);
+
+                if (typeId == null)
+                {
+                    Log.Warn($"Type not found. (Branch: {currentPolicy.Branch}, Type: {policy.Type})");
+                }
+
+                if (tfsPolicies != null)
+                {
+                    foreach (var tfsPolicy in tfsPolicies)
+                    {
+                        if (typeId == tfsPolicy.Type.Id &&
+                                ((string.IsNullOrEmpty(tfsPolicy.GetBranch()) && string.IsNullOrEmpty(policy.Branch)) ||
+                                (!string.IsNullOrEmpty(tfsPolicy.GetBranch()) && !string.IsNullOrEmpty(policy.Branch) &&
+                                    tfsPolicy.GetBranch() == policy.Branch && tfsPolicy.GetMatchKind() == policy.MatchKind)))
+                        {
+                            hasMatch = true;
+                            handledTfsPolicies.Add(tfsPolicy.Id);
+
+                            if (policy.TfsEquals(tfsPolicy))
+                            {
+                                Log.Info($"Policy is uptodate. (Branch: {currentPolicy.Branch}, Type: {policy.Type})");
+                            }
+                            else
+                            {
+                                this.UpdatePolicy(policyClient, types, projectId, repository, currentPolicy, policy, tfsPolicy);
+                            }
+
+                            break;
+                        }
+                    }
+                }
+
+                if (!hasMatch)
+                {
+                    this.CreatePolicy(policyClient, types, projectId, repository, currentPolicy, policy);
+                }
+            }
+        }
+
+        private void RemoveNoMatchTfsPolicies(bool allowDeletion, PolicyHttpClient policyClient, Guid projectId, IEnumerable<PolicyConfiguration> tfsPolicies, List<int> handledTfsPolicies)
```

Comment:
Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server